### PR TITLE
test(examples): add and test histogram sum/count-split configuration

### PR DESCRIPTION
**Pull request by pichlermarc**
This PR adds the config to split `sum`/`count` metrics from Histograms from the [Dynatrace Documentation](https://docs.dynatrace.com/docs/extend-dynatrace/opentelemetry/getting-started/metrics/ingest/migration-guide-otlp-exporter#in-collector-additional-features) as well as tests for that config.

### CODE REVIEW

File: testbed/integration/config_examples_test.go [hunk 1]
```diff
@@ -202,3 +203,113 @@ func TestConfigJaegerGrpc(t *testing.T) {
...
+	histogramDataPoint.SetCount(3)
+	histogramDataPoint.SetSum(7.5)
+	histogramDataPoint.SetMin(0.5)
+	histogramDataPoint.SetMax(3.5)
+	histogramDataPoint.BucketCounts().Append(1, 0, 0, 0, 2)
+
+	// Split Histogram
+	splitCountMetric := expectedMetrics.AppendEmpty()
+	splitCountMetric.SetName("my.histogram_count")
+	splitCountMetric.SetUnit("custom_unit")
```
Comment (pichlermarc) [inline testbed/integration/config_examples_test.go:259]: **Note for reviewers:** does it make sense to have the `unit` be the same for `count` metrics split from `histogram` metrics?

Take [`http.server.request.duration` metric](https://opentelemetry.io/docs/specs/semconv/http/http-metrics/#http-server) for example, it records the duration of requests on the server. The unit for that metric is `s` (seconds), however, the `count` property on that histogram is actually the amount of requests recorded, not seconds (that's the sum, which is the total time the server spent on handling requests of that route, for instance).

Comment (pichlermarc) [reply testbed/integration/config_examples_test.go:259]: I opened an issue upstream as I think it's applicable to all metrics that can be passed to that function: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31575

Comment (pichlermarc) [reply testbed/integration/config_examples_test.go:259]: I suggest we go ahead with adding the tests as-is then iterate on them based on the outcome of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31575 

WDYT @evan-bradley @arminru?

File: testbed/integration/metrics_validation.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+func (v *MetricsSampleConfigsValidator) Validate(tc *testbed.TestCase) {
+	actualMetrics := tc.MockBackend.DataItemsReceived()
+
+	assert.EqualValues(v.t, v.expectedMetrics.MetricCount(), actualMetrics, "Received and expected number of metrics do not match.")
+	assertExpectedMetricsAreInReceived(v.t, []pmetric.Metrics{v.expectedMetrics}, tc.MockBackend.ReceivedMetrics)
+}
+
+func (v *MetricsSampleConfigsValidator) RecordResults(tc *testbed.TestCase) {
+}
```
Comment (pirgeo) [inline testbed/integration/metrics_validation.go:34]: Do we need this?

File: testbed/integration/config_examples_test.go [hunk 3]
```diff
@@ -202,3 +203,113 @@ func TestConfigJaegerGrpc(t *testing.T) {
...
+	tc := testbed.NewTestCase(
+		t,
+		dataProvider,
+		sender,
+		receiver,
+		col,
+		validator,
+		&testbed.CorrectnessResults{},
+	)
+	t.Cleanup(tc.Stop)
```
Comment (pirgeo) [inline testbed/integration/config_examples_test.go:295]: Is this to make sure nothing else is running in the background?

**Review by pirgeo — APPROVED** [review]
Looks good to me. I'll defer to someone with more knowledge of the testbed framework, but from a high level everything looks good!

File: testbed/integration/config_examples_test.go (hunk 3 above)
Comment (pirgeo) [reply testbed/integration/config_examples_test.go:295]: Ah, I see. It's like a `defer` for the test. Got it, thank you

**Pull request merged**